### PR TITLE
ci: test everything when glue code changes

### PR DESCRIPTION
**Pull request by benesch**
@philip-stoev what do you think about this? 

This is an alternative to #5556. The idea is that when anything about
the CI configuration itself changes, we ought not attempt to trim any
jobs from the CI pipeline, and just run everything.

---
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/5560)

### CODE REVIEW

**Review by philip-stoev — APPROVED** [review]
Yes, this should do the trick. Thanks for forcing ```main``` to not enjoy any shortcuts, that would be the safest approach.

File: ci/test/mkpipeline.py [hunk 1]
```diff
@@ -26,18 +26,40 @@
...
+
 
 def main() -> int:
+    # Make sure we have an up to date view of main.
+    spawn.runv(["git", "fetch", "origin", "main"])
+
+    # Print out a summary of all changes.
+    os.environ["GIT_PAGER"] = ""
+    spawn.runv(["git", "diff", "--stat", "origin/main..."])
+
```
Comment (quodlibetor) [inline ci/test/mkpipeline.py:54]: I don't how likely it is to come up in practice, but I think you actually want to diff between HEAD and the merge-base between HEAD and main. I think that as things stand right now you're more likely to be over-agressive in testing too much, including in branches that are forked off before changes to glue code were merged.

Comment (quodlibetor) [reply ci/test/mkpipeline.py:54]: Oh wait that's what three dots does, nevermind.

**Review by quodlibetor — APPROVED** [review]

**Review by cirego — APPROVED** [review]

**Comment by benesch** [comment]
Thanks very much for making the initial PR for this, @philip-stoev!

**Pull request merged**